### PR TITLE
Address issue 47: name clashes in obfuscated types.

### PR DESCRIPTION
**Pull request by jdsheneman**
The fix for the issue is contained in MetaType.Field.cs.  A unit test for the fix is in Issue47_Obfuscation.cs.

The changes to the .sln file are all the result of adding the Obfuscated project, a test target DLL.  This project builds a DLL from the Obfuscated.il, obfuscated to rename an inner class, two methods and a field to the same name "a".

### CODE REVIEW

File: src/protobuf-net/Meta/MetaType.Fields.cs [hunk 1]
```diff
@@ -178,8 +178,24 @@ private ValueMember AddField(int fieldNumber, string memberName, Type itemType,
             if (Type.IsArray) throw new InvalidOperationException("Can't add fields to array type");
             MemberInfo mi = null;
             MemberInfo[] members = Type.GetMember(memberName, Helpers.IsEnum(Type) ? BindingFlags.Static | BindingFlags.Public : BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
-            if (members != null && members.Length == 1) mi = members[0];
-            if (mi == null) throw new ArgumentException("Unable to determine member: " + memberName, nameof(memberName));
+            if (members != null)
+            {
+                if (members.Length == 1)
+                    mi = members[0];
+                else if (members.Length > 1)
```
Comment (AqlaSolutions) [inline src/protobuf-net/Meta/MetaType.Fields.cs]: This is an ambiguous case. You didn't add any logic to determine which one should be used. Picking just any of them is not an acceptable solution.

Comment (AqlaSolutions) [reply src/protobuf-net/Meta/MetaType.Fields.cs]: Will it work if we filter members before the Length check to take only fields and properties?

**Review by AqlaSolutions — CHANGES_REQUESTED** [review]
See comments

**Comment by AqlaSolutions** [comment]
Also please target develop branch instead

File: src/protobuf-net/Meta/MetaType.Fields.cs (hunk 1 above)
Comment (jdsheneman) [reply src/protobuf-net/Meta/MetaType.Fields.cs]: I think it's possible to filter the elements first.

Comment (jdsheneman) [reply src/protobuf-net/Meta/MetaType.Fields.cs]: I'll work on the requested changes.

**Comment by jdsheneman** [comment]
I will reopen this pull request with new changes to a develop branch.

**Pull request closed without merging**